### PR TITLE
[RHOAIENG-2389] Outdated version info for DS Pipelines

### PR DESCRIPTION
**Pull request by andrewballantyne**
https://issues.redhat.com/browse/RHOAIENG-2389
Removes outdated warning as well: https://issues.redhat.com/browse/RHOAIENG-2972

## Description

Removes the deprecation banner from https://github.com/opendatahub-io/odh-dashboard/pull/2468 & updates the v2 deprecation state for v1 projects.

Screenshot 2024-02-12 at 3 51 01 PM
image
Screenshot 2024-02-12 at 3 51 34 PM

cc @yannnz @xianli123 @kywalker-rh 

## How Has This Been Tested?

Manually, installing DSPv1 and then loading up this branch / PR.

## Test Impact

Wrote a test to make sure when the DSPA is installed but outdated the buttons don't appear.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

File: frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts [hunk 1]
```diff
@@ -120,4 +120,26 @@ describe('PipelinesList', () => {
...
+      {
+        method: 'GET',
+        pathname:
+          '/api/k8s/apis/datasciencepipelinesapplications.opendatahub.io/v1alpha1/namespaces/test-project/datasciencepipelinesapplications/dspa',
+      },
+      mockDataSciencePipelineApplicationK8sResource({ dspVersion: 'v1' }),
+    );
+    projectDetails.visit('test-project');
+
+    pipelinesSection.findAllActions().should('have.length', 1);
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelinesList.cy.ts:143]: Good call, the test shifted on me when I was reviewing doc links and looked at the mocks. I need to fix that confusion

File: frontend/src/pages/projects/screens/detail/DetailsSection.tsx [hunk 2]
```diff
@@ -76,7 +77,11 @@ const DetailsSection: React.FC<DetailsSectionProps> = ({
             </Title>
           </FlexItem>
           {actions && (
-            <Flex direction={{ default: 'row' }} spaceItems={{ default: 'spaceItemsSm' }}>
+            <Flex
+              data-testId="details-section-action"
```
[the remark this answers is not in the document]
Comment (andrewballantyne) [reply frontend/src/pages/projects/screens/detail/DetailsSection.tsx:81]: oh doh -- that's on the wrong element 🤦 thanks for catching it -- it was intended on the `FlexItem` -- rushed it in when I noticed it this morning before my meetings

**Review by jpuzz0 — APPROVED** [review]
/lgtm

**Pull request merged**